cmd/create/accountroles/cmd.go

```diff
@@ -382,32 +405,38 @@ func createRoles(r *rosa.Runtime, prefix, permissionsBoundary, accountID, env st
 			"aws_account_id": aws.GetJumpAccount(env),
 		})
 		r.Reporter.Debugf("Creating role '%s'", accRoleName)
+		tagsList := map[string]string{
+			tags.OpenShiftVersion: defaultPolicyVersion,
+			tags.RolePrefix:       prefix,
+			tags.RoleType:         file,
+			tags.RedHatManaged:    "true"}
+		if managedPolicies {
+			tagsList[tags.ManagedPolicies] = "true"
+		}
 		roleARN, err := r.AWSClient.EnsureRole(accRoleName, policy, permissionsBoundary,
-			defaultPolicyVersion, map[string]string{
-				tags.OpenShiftVersion: defaultPolicyVersion,
-				tags.RolePrefix:       prefix,
-				tags.RoleType:         file,
-				tags.RedHatManaged:    "true",
-			}, path)
+			defaultPolicyVersion, tagsList, path, managedPolicies)
 		if err != nil {
 			return err
 		}
 		r.Reporter.Infof("Created role '%s' with ARN '%s'", accRoleName, roleARN)
 
 		filename = fmt.Sprintf("sts_%s_permission_policy", file)
-		policyDetail = aws.GetSTSPolicyDetails(policies, filename)
-
-		r.Reporter.Debugf("Creating permission policy '%s'", policyARN)
-		policyARN, err = r.AWSClient.EnsurePolicy(policyARN, policyDetail,
-			defaultPolicyVersion, map[string]string{
-				tags.OpenShiftVersion: defaultPolicyVersion,
-				tags.RolePrefix:       prefix,
-				tags.RoleType:         file,
-				tags.RedHatManaged:    "true",
-			}, path)
-		if err != nil {
-			return err
+		if managedPolicies {
+			policyARN, err = aws.GetSTSPolicyARN(policies, filename)
+			if err != nil {
+				return err
+			}
+		} else {
+			policyDetail = aws.GetSTSPolicyDetails(policies, filename)
```

Comment:
this var should be used inside this block only

Reply:
Fixed.